deepreg/model/layer.py

```diff
@@ -1,451 +1,458 @@
-        :param inputs: shape = [batch, dim1, dim2, dim3, channels]
-        :param kwargs:
-        :return:
-        """
-        if inputs.shape[4] % self._stride != 0:
-            raise ValueError("The channel dimension can not be divided by the stride")
-        output = self._resize3d(inputs=inputs)
-        output = tf.split(output,
-                          num_or_size_splits=self._stride,
-                          axis=4)  # a list of [batch, out_dim1, out_dim2, out_dim3, channels//stride], num = stride
-        output = tf.reduce_sum(tf.stack(output, axis=5),
-                               axis=5)  # [batch, out_dim1, out_dim2, out_dim3, channels//stride]
-        return output
-
-
-class LocalNetResidual3dBlock(tf.keras.layers.Layer):
-    def __init__(self, filters, kernel_size=3, strides=1, **kwargs):
-        super(LocalNetResidual3dBlock, self).__init__(**kwargs)
-        # init layer variables
-        self._conv3d = Conv3d(filters=filters, kernel_size=kernel_size, strides=strides, use_bias=False)
-        self._norm = Norm()
-        self._act = Activation()
-
-    def call(self, inputs, training=None, **kwargs):
-        layer_util.check_inputs(inputs, 2, "ResidualBlock")
-
-        return self._act(self._norm(inputs=self._conv3d(inputs=inputs[0]),
-                                    training=training) + inputs[1])
-
-
-class LocalNetUpSampleResnetBlock(tf.keras.layers.Layer):
-    def __init__(self, filters, use_additive_upsampling=True, **kwargs):
-        super(LocalNetUpSampleResnetBlock, self).__init__(**kwargs)
-        # save parameters
-        self._filters = filters
-        self._use_additive_upsampling = use_additive_upsampling
-        # init layer variables
-        self._deconv3d_block = None
-        self._additive_upsampling = None
-        self._conv3d_block = Conv3dBlock(filters=filters)
-        self._residual_block = LocalNetResidual3dBlock(filters=filters, strides=1)
-
-    def build(self, input_shape):
-        super(LocalNetUpSampleResnetBlock, self).build(input_shape)
-        layer_util.check_inputs(input_shape, 2, "UpSampleResnetBlock build")
-
-        output_shape = input_shape[1][1:4]
-        self._deconv3d_block = Deconv3dBlock(filters=self._filters, output_shape=output_shape, strides=2)
-        if self._use_additive_upsampling:
-            self._additive_upsampling = AdditiveUpSampling(output_shape=output_shape)
-
-    def call(self, inputs, training=None, **kwargs):
-        layer_util.check_inputs(inputs, 2, "UpSampleResnetBlock call")
-
-        inputs_nonskip, inputs_skip = inputs[0], inputs[1]
-        h0 = self._deconv3d_block(inputs=inputs_nonskip, training=training)
-        if self._use_additive_upsampling:
-            h0 += self._additive_upsampling(inputs=inputs_nonskip)
-        r1 = h0 + inputs_skip
-        r2 = self._conv3d_block(inputs=h0, training=training)
-        h1 = self._residual_block(inputs=[r2, r1], training=training)
-        return h1
+import tensorflow as tf

```

Comment:
I think some spaces have been added to this file?

Reply:
is it a problem?

Reply:
You can delete them next time you merge 😊